### PR TITLE
Endpoint for new chat messages

### PR DESCRIPTION
**Pull request by petterboussard**
Endpoint to post new chat messages.

See more at README.md file.

This pull request is for the implementation of the back-end tast, https://trello.com/c/jkpe2mqg/16-endpoint-para-agregar-un-mensaje-a-un-chat , for the story card https://trello.com/c/GRYbvTEh/34-como-usuario-quiero-comunicarme-con-un-miembro-de-soporte-para-poder-resolver-un-problema

### CODE REVIEW

File: app/controllers/api/v1/chats_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    else
+      chat = Chat.create
+      create_message message_params[:body], chat
+    end
+  end
+  # rubocop:enable Metrics/PerceivedComplexity
+
+  private
+
+  def create_message(body, chat, user_id = nil)
```
Comment (jicc96) [inline app/controllers/api/v1/chats_controller.rb]: You should not create a chat. If the chat given has a parameter does not exists then the function should return a error message. The user from the widget has an endpoint function to create a chat

File: README.md [hunk 2]
```diff
@@ -1 +1,15 @@
...
+
+### Post a new message:  
+```
+post app_domain/api/v1/chats/new_message
+```
+> This endpoint is intended to be used from the widget and the dashboard.  
+If it is being called from the widget you must provide a user_id in the request body squema as well as the chat_id.  
+You can get the user_id by calling the sign-in endoint and it should return all user attrs.  
+If it is being called from the widget there are two scenarios: 
+* It is the first message, then you should provide just the body of the message and the response will provide you with the chat_id created for the message.
```
[the remark this answers is not in the document]
Comment (petterboussard) [reply README.md]: Updated!

File: README.md [hunk 3]
```diff
@@ -1 +1,15 @@
-# pis2019-chat-dashboard
\ No newline at end of file
+# API Endpoints: 
+
+### Post a new message:  
+```
+post app_domain/api/v1/chats/new_message
+```
+> This endpoint is intended to be used from the widget and the dashboard.  
+If it is being called from the widget you must provide a user_id in the request body squema as well as the chat_id.
```
[the remark this answers is not in the document]
Comment (petterboussard) [reply README.md]: Updated!

File: config/routes.rb [hunk 4]
```diff
@@ -2,9 +2,19 @@
...
   mount ActionCable.server => '/cable'
+
+  defaults format: :json do 
+  	namespace :api do 
+  		namespace :v1 do 
+  			post 'chats/new_message', to: 'chats#new_message'
+  			post 'chats/new_dashboard_message', to: 'chats#new_dashboard_message'
+  		end 
+  	end 
+	end
```
Comment (jicc96) [inline config/routes.rb]: Bad indentation this end should be 2 spaces to the left

**Pull request merged**